src/cpu/pred/bpred_unit.cc

```diff
@@ -503,10 +503,12 @@ BPredUnit::squash(const InstSeqNum &squashed_sn,
                         "return [sn:%llu] PC: %#x Restoring RAS\n", tid,
                         squashed_sn,
                         hist_it->seqNum, hist_it->pc);
-                DPRINTF(Branch,
-                        "[tid:%i] [squash sn:%llu] Restoring top of RAS "
-                        "to: %i, target: %s\n", tid, squashed_sn,
-                        hist_it->RASIndex, *hist_it->RASTarget);
+                if (hist_it->RASTarget != nullptr) {
+                    DPRINTF(Branch,
+                            "[tid:%i] [squash sn:%llu] Restoring top of RAS "
+                            "to: %i, target: %s\n", tid, squashed_sn,
+                            hist_it->RASIndex, *hist_it->RASTarget.get());
+                }
```

Comment:
Out of curiosity, why not something like this? Doesn't it make sense to print even if there is no `RASTarget`?

```suggestion
                 DPRINTF(Branch,
                        "[tid:%i] [squash sn:%llu] Restoring top of RAS "
                        "to: %i, target: %s\n", tid, squashed_sn,
                        hist_it->RASIndex, 
                        hist_it->RASIndex ? *hist_it->RASTarget.get() : "no RAS");
```

Reply:
Thanks for your suggestion. I'm not familiar with the part for cpu branch prediction, any changes make --debug-flags=Branch will LGTM.